### PR TITLE
7800v2 can driver add

### PR DESCRIPTION
**Pull request by luckshot**
Custom CAN driver for TS-7800-V2 added.

### CODE REVIEW

File: drivers/net/can/sja1000/Kconfig [hunk 1]
```diff
@@ -102,6 +102,12 @@ config CAN_SJA1000_PLATFORM
 	  on embedded systems with OpenFirmware bindings, e.g. if you
 	  have a PowerPC based system you may want to enable this option.
 
+config CAN_TS7800V2
+	tristate "TS-7800v2 FPGA CAN Controller"
```
Comment (ts-kris) [inline drivers/net/can/sja1000/Kconfig]: `tristate "TS-7800v2 FPGA CAN Controller"`
should be
`tristate "TS-7800-V2 FPGA CAN Controller"` to match product number

File: drivers/net/can/sja1000/Kconfig [hunk 2]
```diff
@@ -102,6 +102,12 @@ config CAN_SJA1000_PLATFORM
...
+	help
+	  Say Y here if you have a Technologic Systems TS-7800v2 board and want to
```
Comment (ts-kris) [inline drivers/net/can/sja1000/Kconfig]: Same as above

File: drivers/net/can/sja1000/ts7800v2_can.c [hunk 3]
```diff
@@ -0,0 +1,287 @@
+/*
+ * ts7800v2_can.c - Interface for the CAN controller in the TS-7800v2's FPGA.
+ *
+ * Copyright (c) 2018, Technologic Systems, Inc.
```
Comment (ts-kris) [inline drivers/net/can/sja1000/ts7800v2_can.c]: Use our updated copyright string. If no major changes to the driver when it was imported here, the date can stay the same.

File: drivers/net/can/sja1000/ts7800v2_can.c [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+   while(readl(&syscon[CAN_CNTL_REG_OFFSET / 4]) & CAN_CNTL_START);
+
+   writel(CAN_CNTL_START | reg, &syscon[CAN_CNTL_REG_OFFSET / 4]);
+
+   do {
+      v = readl(&syscon[CAN_CNTL_REG_OFFSET / 4]);
+   } while(v & CAN_CNTL_START);
+
+#if (DEBUG_ON)
```
Comment (ts-kris) [inline drivers/net/can/sja1000/ts7800v2_can.c]: I'd assume we want all instances of `DEBUG_ON` to follow kernel paradigm of just `DEBUG` but I'm not sure where `DEBUG_ON` originated.

File: drivers/net/can/sja1000/ts7800v2_can.c [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+   if (IS_ERR(addr)) {
+      ret = PTR_ERR(addr);
+      return -ENOMEM;
+   }
+
+	irq = platform_get_resource(pdev, IORESOURCE_IRQ, 0);
+	if (!irq) {
+		dev_err(&pdev->dev, "failed to find IRQ\n");
+		return -ENXIO;
+   }
```
Comment (ts-kris) [inline drivers/net/can/sja1000/ts7800v2_can.c]: Looks like tabs got screwed up here

File: drivers/net/can/sja1000/ts7800v2_can.c [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+   .remove	= ts7800v2_can_remove,
+   .driver	= {
+      .name = DRIVER_NAME,
+      .of_match_table	= ts7800v2_can_of_match,
+   },
+};
+
+module_platform_driver(ts7800v2_can_driver);
+
+MODULE_DESCRIPTION("TS7800v2 CAN interface");
```
Comment (ts-kris) [inline drivers/net/can/sja1000/ts7800v2_can.c]: Change to `TS-7800-V2`

File: drivers/net/can/sja1000/ts7800v2_can.c [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Technologic Systems, Inc");
```
Comment (ts-kris) [inline drivers/net/can/sja1000/ts7800v2_can.c]: Have this be the actual human who wrote this

**Review by ts-kris — CHANGES_REQUESTED** [review]
Also, please squash the latter two commits so we don't end up with a commit with a broken defconfig.

In the future, please use common kernel commit notation, e.g.
`gpio: ts7800: Fix comments`
`can: sja1000: Add ts7800v2 support`
Essentially call out the driver layers until you get to the actual issue. See `git log` of files in the same folder for hints as to how it should look. If you want to amend the commit comments when you squash above, great.

File: drivers/net/can/sja1000/ts7800v2_can.c (hunk 4 above)
Comment (luckshot) [reply drivers/net/can/sja1000/ts7800v2_can.c]: Maybe @markfeathers can comment on this?

**Comment by ts-kris** [comment]
Everything looks good to me now. Note that I did no testing on this.

**Pull request merged**